packages/lesmis/src/components/ApproveButton.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import CheckIcon from '@material-ui/icons/Check';
+import { IconButton } from '@tupaia/admin-panel/lib';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import MuiSnackbar from '@material-ui/core/Snackbar';
+import { SmallAlert } from '@tupaia/ui-components';
+import { useApproveSurveyResponse } from '../api/mutations/useApproveSurveyResponse';
+import { FlexCenter } from './Layout';
+import { GREEN } from '../constants';
+
+const Button = styled(IconButton)`
+  width: 56px;
+`;
+
+export const ApproveButton = ({ value: id }) => {
+  const [isApproved, setIsApproved] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+  const { mutate, isLoading, isError, isSuccess } = useApproveSurveyResponse();
+
+  const handleClose = () => {
+    setShowAlert(false);
+  };
+
+  const handleClickAccept = async () => {
+    mutate(id, {
```

Comment:
I find the word `mutate` a little generic here, but is it the convention for this type of custom hook?

Reply:
It's the plain react-query function rather than anything I've written so it is a convention in the react-query context but it could also make sense to name it to a more local context. I don't mind either way.